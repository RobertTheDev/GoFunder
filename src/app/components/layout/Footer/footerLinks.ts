const footerLinks: {
    href: string;
    name: string;
}[] = [
    {
        href: '/about',
        name: 'About'
    },
    { href: '/accessibility-statement', name: 'Accessibility Statement' },
    { href: '/privacy-policy', name: 'Privacy Policy' },
    { href: '/terms-and-conditions', name: 'Terms and Conditions' }
];

export default footerLinks;
