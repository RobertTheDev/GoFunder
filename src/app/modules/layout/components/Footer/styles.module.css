.footerContainer {
    align-items: center;
    background-color: white;
    display: flex;
    flex-direction: column;
    gap: 3.2rem;
    justify-content: center;
    padding: 3.2rem;
    width: 100%;

    @media (width >=1024px) {
        flex-direction: row;
        gap: 6.4rem;
        padding: 2.4rem 3.2rem;
    }

    @media (prefers-color-scheme: dark) {
        background-color: var(--neutral900);
    }
}

.footerCopyrightText {
    color: var(--neutral900);
    font-size: 1.4rem;

    @media (prefers-color-scheme: dark) {
        color: var(--neutral50);
    }
}

.footerLinkActive {
    color: #00a95c;
    font-size: 1.2rem;
    font-weight: 500;
    text-decoration: none;
}

.footerLinkMenu {
    align-items: center;
    display: flex;
    flex-direction: column;
    gap: 3.2rem;
    justify-content: center;

    @media (width >=1024px) {
        flex-direction: row;
        gap: 6.4rem;
    }
}

.footerLinkText {
    color: var(--neutral900);
    font-size: 1.4rem;
    text-decoration: none;

    @media (prefers-color-scheme: dark) {
        color: var(--neutral50);
    }
}