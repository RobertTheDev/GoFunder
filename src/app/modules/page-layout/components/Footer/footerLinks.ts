export const footerLinks = [
    { href: "/info/accessibility-statement", name: "Accessibility Statement" },
    { href: "/info/privacy-policy", name: "Privacy Policy" },
    { href: "/info/terms-and-conditions", name: "Terms and Conditions" },
];
