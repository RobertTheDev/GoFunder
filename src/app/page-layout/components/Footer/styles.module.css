.footerContainer {
    align-items: center;
    background-color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
    padding: 3.2rem;
    gap: 3.2rem;

    @media (width >=1024px) {
        flex-direction: row;
        gap: 3.2rem;
        padding: 2.4rem 3.2rem;
    }

    @media (prefers-color-scheme: dark) {
        background-color: var(--slate-900);
        color: var(--slate-300);
    }
}

.footerCopyrightText {
    color: var(--neutral-700);
    font-size: 1.4rem;
    font-weight: 400;

    @media (prefers-color-scheme: dark) {
        color: var(--neutral50);
    }
}

.footerLinkActive {
    color: #00a95c;
    font-size: 1.4rem;
    font-weight: 400;
    text-decoration: none;
}

.footerLinkMenu {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    gap: 3.2rem;

    @media (width >=1024px) {
        flex-direction: row;

    }
}

.footerLinkText {
    font-size: 1.4rem;
    font-weight: 400;
    text-decoration: none;

    @media (prefers-color-scheme: dark) {
        color: var(--slate-300);
    }
}

.footerLinkText:hover {
    text-decoration: underline;
}