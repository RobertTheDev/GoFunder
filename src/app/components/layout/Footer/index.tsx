import Link from 'next/link';
import { usePathname } from 'next/navigation';
import footerLinks from './footerLinks';

export default function Footer() {
    const pathName = usePathname();

    return (
        <footer className="footer-container">
            <span className="footer-copyright">
                Â©{new Date().getFullYear()} GoFunder
            </span>
            <nav className="footer-menu">
                {footerLinks.map(link => (
                    <Link
                        href={link.href}
                        key={link.href}
                        className={
                            pathName === link.href
                                ? 'footer-link-active'
                                : 'footer-link'
                        }
                    >
                        {link.name}
                    </Link>
                ))}
            </nav>
        </footer>
    );
}
