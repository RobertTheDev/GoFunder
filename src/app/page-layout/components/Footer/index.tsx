"use client";

import Link from "next/link";
import { usePathname } from "next/navigation";
import footerLinks from "./footerLinks";
import styles from "./styles.module.css";

export default function Footer() {
    const pathName = usePathname();

    return (
        <footer className={styles.footerContainer}>
            <nav className={styles.footerLinkMenu}>
                {footerLinks.map((link) => (
                    <Link
                        href={link.href}
                        key={link.href}
                        className={
                            pathName === link.href
                                ? styles.footerLinkActive
                                : styles.footerLinkText
                        }
                    >
                        {link.name}
                    </Link>
                ))}
            </nav>
            <small className={styles.footerCopyrightText}>
                Â©{new Date().getFullYear()} GoFunder
            </small>
        </footer>
    );
}
