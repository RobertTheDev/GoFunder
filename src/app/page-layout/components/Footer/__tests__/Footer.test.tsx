import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Footer from "../index";

describe("Footer component", () => {
    it("renders without crashing", () => {
        render(<Footer />);
    });

    it("matches snapshot", () => {
        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("displays correct number of links", () => {
        const { getAllByRole } = render(<Footer />);
        const links = getAllByRole("link");
        expect(links).toHaveLength(3);
    });

    it("displays correct copyright year", () => {
        const { getByText } = render(<Footer />);
        const currentYear = new Date().getFullYear();
        expect(getByText(`Â©${currentYear} GoFunder`)).toBeInTheDocument();
    });
});
