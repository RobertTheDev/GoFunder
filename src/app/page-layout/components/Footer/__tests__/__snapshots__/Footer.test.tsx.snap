// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="_footerContainer_fd86be"
  >
    <nav
      class="_footerLinkMenu_fd86be"
    >
      <a
        class="_footerLinkText_fd86be"
        href="/info/accessibility-statement"
      >
        Accessibility Statement
      </a>
      <a
        class="_footerLinkText_fd86be"
        href="/info/privacy-policy"
      >
        Privacy Policy
      </a>
      <a
        class="_footerLinkText_fd86be"
        href="/info/terms-and-conditions"
      >
        Terms and Conditions
      </a>
    </nav>
    <small
      class="_footerCopyrightText_fd86be"
    >
      Â©2024 GoFunder
    </small>
  </footer>
</DocumentFragment>
`;
